lib/ocpp/v201/charge_point.cpp

```diff
@@ -436,7 +456,33 @@ void ChargePoint::init_websocket() {
                           << " which is configurationSlot: " << configuration_slot;
             this->websocket_connection_status = WebsocketConnectionStatusEnum::Disconnected;
             this->message_queue->pause();
+            // B04.FR.01
+            // If offline period exceeds Offline threshold, send status notification for all all connectors.
+            int offline_threshold = this->device_model->get_value<int>(ControllerComponentVariables::OfflineThreshold);
+            if (offline_threshold > 0) {
+
+                // get the status of all the connectors
+                for (auto const& [evse_id, evse] : this->evses) {
+                    int number_of_connectors = evse->get_number_of_connectors();
+                    conn_states_per_evse_id_key conn_states_struct_key;
+                    conn_states_per_evse_id_value conn_states_struct_value;
+                    EVLOG_debug<< "LIBOCPP: evseId: " << evse_id << " numConn: " << number_of_connectors;
+                    for (int connector_id = 1; connector_id <= number_of_connectors; connector_id++) {
```

Comment:
I don't like the fact that you need to iterate the connectors here manually. I know this is currently the only option, but maybe we can improve this by adding a getter function for the entire connector map of an Evse?